clients/csharp-uwp/UWPVoiceAssistantSample/KeywordRegistration.cs

```diff
@@ -344,17 +317,46 @@ private async Task SetModelDataIfNeededAsync(ActivationSignalDetectionConfigurat
                 // Update was successful. Record this so we don't repeat it needlessly!
                 this.LastUpdatedActivationKeywordModelVersion = this.AvailableActivationKeywordModelVersion;
 
-                // And now, re-enable the configuration.
-                await configuration.SetEnabledAsync(true);
+                // And now, re-enable the configuration if we previously disabled it.
+                if (configurationWasEnabled)
+                {
+                    await configuration.SetEnabledAsync(true);
+                }
             }
-        }
+        }
+
+        private async Task<ActivationSignalDetectionConfiguration> CreateKeywordConfigurationAsyncInternal()
+        {
+            var detector = await GetFirstEligibleDetectorAsync(this.KeywordActivationModelDataFormat);
 
-        //private async Task<StorageFile> GetFileFromPathAsync(string path)
-        //    => path.StartsWith("ms-appx", StringComparison.InvariantCultureIgnoreCase)
-        //        ? await StorageFile.GetFileFromApplicationUriAsync(new Uri(path))
-        //        : await StorageFile.GetFileFromPathAsync(path);
+            // Only one configuration may be active at a time. Before creating a new one, ensure all existing ones
+            // are disabled to avoid collisions.
+            var configurations = await detector.GetConfigurationsAsync();
+            foreach (var configuration in configurations)
+            {
+                await configuration.SetEnabledAsync(false);
+            }
+
+            var targetConfiguration = await GetOrCreateConfigurationOnDetectorAsync(
+                detector,
+                this.KeywordDisplayName,
+                this.KeywordId,
+                this.KeywordModelId);
+            await this.SetModelDataIfNeededAsync(targetConfiguration);
+
+            if (!targetConfiguration.IsActive)
+            {
+                await targetConfiguration.SetEnabledAsync(true);
+            }
+
+            this.keywordConfiguration = targetConfiguration;
+
+            return targetConfiguration;
+        }
 
         private async Task<StorageFile> GetFileFromPathAsync(string path)
```

Comment:
Notes on this one:
* there's no need to copy packaged models into appdata
* we can resolve ms-appx and ms-appdata URIs
* the helper also resolves non-ms-app* paths